library/core/src/ascii.rs

```diff
@@ -138,7 +138,7 @@ impl FusedIterator for EscapeDefault {}
 impl fmt::Display for EscapeDefault {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         // SAFETY: ok because `escape_default` created only valid utf-8 data
-        f.write_str(unsafe { from_utf8_unchecked(&self.data[self.range.clone()]) })
+        f.write_str(unsafe { from_utf8_unchecked(&self.data[..self.range.end as usize]) })
```

Comment:
This is a change in behavior if the iterator has already been advanced from the front, right?

Reply:
Nope, when dropped second commit noticed, that i changed this in first.